src/CoreWCF.Primitives/src/CoreWCF/Description/ServiceAuthorizationBehavior.cs

```diff
@@ -112,8 +112,7 @@ public bool ImpersonateCallerForAllOperations
             }
             set
             {
-                ThrowIfImmutable();
-                impersonateCallerForAllOperations = value;
```

Comment:
Can you leave the old code commented like you did below so it's obvious what should be here. Does impersonation not work? I put some effort in to making impersonation work with different api's now.

Reply:
Will write a test to see if it works